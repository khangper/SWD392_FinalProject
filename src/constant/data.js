export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
