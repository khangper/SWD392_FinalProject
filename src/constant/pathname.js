export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
