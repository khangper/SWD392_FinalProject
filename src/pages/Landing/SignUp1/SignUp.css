.SignUp-input-wrapper {
    position: relative;
    width: 100%;
    margin-bottom: 20px;
}

.SignUp-input-field {
    width: 80%;
    padding: 10px 10px 10px 10px;
    border: 1px solid #ddd;
    border-radius: 4px;
    font-size: 14px;
}

.loginSingUp-btn {
    font-size: 14px;
    margin-top: 30px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #fff;
    background: #ed2a26;
    border-radius: 3px;
    text-align: center;
    border: 0;
    width: 80%;
    height: 40px;
}

.loginSingUp-btn:hover {
    color: #fff;
    background: #333;
}

.By-signingup {
    padding: 47px 0 30px;
    margin-bottom: 0 !important;
    border-bottom: 1px solid #efefef;
}

.By-signingup a {
    color: #ed2a26;
}

.By-signingup a:hover {
    color: #333 !important;
    text-decoration: underline !important;
}

.login-move {
    width: 38.7px;
    position: relative;
    font-size: 14px;
    line-height: 26px;
    font-family: Roboto;
    color: #4183c4;
    text-align: center;
}

.LoginMove {
    width: 38.7px;
    position: relative;
    font-size: 14px;
    line-height: 26px;
    font-family: Roboto;
    color: #4183c4;
    text-align: center;
}

.remember-meSignUp {
    display: flex;
    align-items: center;
    margin-bottom: 20px;
    margin-left: auto; /* Di chuyển khối qua bên phải */
    margin-right: 9px;
}

.Label-signup {
    text-align: left;
}

.remember-meSignUp input[type="checkbox"] {
    display: none;
    padding-left: 94px;
}

.remember-meSignUp label {
    position: relative;
    padding-left: 94px; /* Space for the custom checkbox */
    cursor: pointer;
    color: rgb(21, 21, 21);
    font-size: 14px;
    user-select: none;
    display: inline-block;
    margin-left: 0; /* Ensure no extra margin on the left */
}

.remember-meSignUp label:before {
    content: '';
    position: absolute;
    left: 61px;
    top: 50%;
    transform: translateY(-50%);
    width: 15px;
    height: 15px;
    border: 1px solid gray;
    border-radius: 50%;
    background-color: white;
}

.remember-meSignUp input[type="checkbox"]:checked + label:before {
    border-color: red;
    background-color: rgb(255, 255, 255);
}

.remember-meSignUp label:after {
    content: '';
    position: absolute;
    left: 68px;
    top: 50%;
    transform: translateY(-50%) rotate(45deg);
    width: 2px;
    height: 7px;
    border: solid red;
    border-width: 0 3px 3px 0;
    opacity: 0;
}

.remember-meSignUp input[type="checkbox"]:checked + label:after {
    opacity: 1;
}
