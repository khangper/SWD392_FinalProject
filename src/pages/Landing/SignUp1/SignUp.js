import React from 'react';
import './SignUp.css';
import { Link } from 'react-router-dom';

export default function SignUp() {
  return (
    <div className="login-container">
      <div className="login-logo-container">
        <Link to='/home'>
          <img src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg" alt="Cursus Logo" className="login-logo" />
        </Link>
      </div>
      <div className="login-form-container">
        <h2>Welcome Back</h2>
        <p className="login-banner">Sign Up and start learning</p>
        <div className="SignUp-input-wrapper">
          <input type="text" placeholder="Full Name" className="SignUp-input-field" />
        </div>
        <div className="SignUp-input-wrapper">
          <input type="email" placeholder="Email Address" className="SignUp-input-field" />
        </div>
        <div className="SignUp-input-wrapper">
          <input type="password" placeholder="Password" className="SignUp-input-field" />
        </div>
        <div className="remember-meSignUp">
          <input type="checkbox" id="emailOptIn" />
          <label htmlFor="emailOptIn" className="Label-signup">
            I’m in for emails with exciting discounts and personalized recommendations
          </label>
        </div>
        <Link to="/StepSignUp">
          <button className="loginSingUp-btn" type="submit">Next</button>
        </Link>
        <p className="By-signingup">
          By signing up, you agree to our <a href="terms_of_use.html">Terms of Use</a> and <a href="terms_of_use.html">Privacy Policy</a>.
        </p>
        <p className="mb-0 mt-30">
          Already have an account? <Link to="/Login" className="LoginMove">Log In</Link>
        </p>
      </div>
      <div className="form-footer">
        <img alt="" className="footer-logo" src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"></img>
        <span>© 2024 <strong>Cursus</strong>. All Rights Reserved.</span>
      </div>
    </div>
  );
}
