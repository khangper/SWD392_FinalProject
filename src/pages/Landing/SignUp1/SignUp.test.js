import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import SignUp from "../../Landing/SignUp1/SignUp";
import "@testing-library/jest-dom/extend-expect";
import { renderWithProviders } from "../../../constant/test-util";

describe("SignUp Component", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });

  it("renders the SignUp component", () => {
    const { container } = renderWithProviders(<SignUp />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("validates name input - allows only letters and spaces", () => {
    renderWithProviders(<SignUp />);
    const nameInput = screen.getByPlaceholderText("Full Name");

    fireEvent.change(nameInput, { target: { value: "John Doe" } });
    expect(
      screen.queryByText("Name cannot contain special characters.")
    ).toBeNull();

    fireEvent.change(nameInput, { target: { value: "John@Doe" } });
    expect(
      screen.getByText("Name cannot contain special characters.")
    ).toBeInTheDocument();
  });

  it("enables the Next button when the name is valid", () => {
    renderWithProviders(<SignUp />);
    const nameInput = screen.getByPlaceholderText("Full Name");
    const nextButton = screen.getByText("Next");

    fireEvent.change(nameInput, { target: { value: "John Doe" } });
    expect(nextButton).not.toBeDisabled();

    fireEvent.change(nameInput, { target: { value: "John@Doe" } });
    expect(nextButton).toBeDisabled();
  });

  it("displays an error message when the name contains special characters", () => {
    renderWithProviders(<SignUp />);
    const nameInput = screen.getByPlaceholderText("Full Name");

    fireEvent.change(nameInput, { target: { value: "John@Doe" } });
    expect(
      screen.getByText("Name cannot contain special characters.")
    ).toBeInTheDocument();

    fireEvent.change(nameInput, { target: { value: "John Doe" } });
    expect(
      screen.queryByText("Name cannot contain special characters.")
    ).toBeNull();
  });
});
