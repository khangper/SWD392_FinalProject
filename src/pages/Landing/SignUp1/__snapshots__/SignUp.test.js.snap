// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders the SignUp component 1`] = `
<div>
  <div
    class="login-container"
  >
    <div
      class="login-logo-container"
    >
      <a
        href="/home"
      >
        <img
          alt="Cursus Logo"
          class="login-logo"
          src="https://gambolthemes.net/html-items/cursus-new-demo/images/logo.svg"
        />
      </a>
    </div>
    <div
      class="login-form-container"
    >
      <h2>
        Welcome Back
      </h2>
      <p
        class="login-banner"
      >
        Sign Up and start learning
      </p>
      <div
        class="SignUp-input-wrapper"
      >
        <input
          class="SignUp-input-field"
          placeholder="Full Name"
          type="text"
          value=""
        />
      </div>
      <div
        class="SignUp-input-wrapper"
      >
        <input
          class="SignUp-input-field"
          placeholder="Email Address"
          type="email"
        />
      </div>
      <div
        class="SignUp-input-wrapper"
      >
        <input
          class="SignUp-input-field"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        class="remember-meSignUp"
      >
        <input
          id="emailOptIn"
          type="checkbox"
        />
        <label
          class="Label-signup"
          for="emailOptIn"
        >
          I’m in for emails with exciting discounts and personalized recommendations
        </label>
      </div>
      <a
        href="/SignUpStep"
      >
        <button
          class="loginSingUp-btn"
          type="submit"
        >
          Next
        </button>
      </a>
      <p
        class="By-signingup"
      >
        By signing up, you agree to our
         
        <a
          href="terms_of_use.html"
        >
          Terms of Use
        </a>
         and
         
        <a
          href="terms_of_use.html"
        >
          Privacy Policy
        </a>
        .
      </p>
      <p
        class="mb-0 mt-30"
      >
        Already have an account?
         
        <a
          class="LoginMove"
          href="/login"
        >
          Log In
        </a>
      </p>
    </div>
    <div
      class="form-footer"
    >
      <img
        alt=""
        class="footer-logo"
        src="https://gambolthemes.net/html-items/cursus-new-demo/images/sign_logo.png"
      />
      <span>
        © 2024 
        <strong>
          Cursus
        </strong>
        . All Rights Reserved.
      </span>
    </div>
  </div>
</div>
`;
